rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/compact/Tile.java

```diff
@@ -299,9 +301,21 @@ private void highlightGroup(boolean selected) {
         if (host != null || nodesource != null) {
             int id;
             if (host == null) {
-                id = panel.indexOf(nodesource).get();
+                final Optional<Integer> opIndexOfNodeSource = panel.indexOf(nodesource);
+                if (opIndexOfNodeSource.isPresent()) {
+                    id = opIndexOfNodeSource.get();
+                } else {
+                    nodesource = null;
+                    return;
+                }
             } else {
-                id = panel.indexOf(host).get();
+                final Optional<Integer> opIndexOfHost = panel.indexOf(host);
+                if (opIndexOfHost.isPresent()) {
+                    id = opIndexOfHost.get();
+                } else {
+                    host = null;
+                    return;
+                }
             }
 
             while (true) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)